virtio-devices/src/pmem.rs

```diff
@@ -424,6 +425,7 @@ impl VirtioDevice for Pmem {
                 move || {
                     if let Err(e) = handler.run(paused, paused_sync.unwrap()) {
                         error!("Error running worker: {:?}", e);
+                        exit_evt.write(1).ok();
```

Comment:
You don't need to add this line here or in any other virtio device. Instead you should modify the following code: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/virtio-devices/src/thread_helper.rs#L47-L52 so that it both checks for panics and errors.

Reply:
Right. Took a cheaper route yesterday. PR is updated which also cleaned-up a bit the use of `spawn_virtio_thread()`.